codice/ddf
# DDF-4141 Fix the table view so it exports the correct number of results in the correct order

#### What does this PR do?
- Sets `count` on the query request for table exports so the right number of results are exported for both `All` and `Visible`.
- Sets `sorts` on the query request for table exports so the results are exported in the right order.
- Displays a warning when a user attempts to export a large number of results.

#### Who is reviewing it? 
@leo-sakh 
@Corey-Collins 
@bellcc 
@glenhein 

#### Select relevant component teams: 
@codice/ui 

#### Ask 2 committers to review/merge the PR and tag them here.
@andrewkfiedler
@djblue 

#### How should this be tested?
1. Ingest a large number of records (at least 250).
2. Open the table view in Intrigue and run a search.
3. Click `Export`.
4. Select `Visible` and any file type and click `Download`.
5. Verify that the exported file contains only the results from the first page in the same order in which they are displayed in Intrigue.
a. Do this for a search that returns at least one full page of results and for a search that returns less than a full page of results.
6. Select `All` and any file type and click `Download`.
7. Verify that the exported file contains as many results as the search card says and that the results are in the same order in which they are displayed in Intrigue.
a. Make sure to test this for a search that returns over 250 results and verify that a warning message is displayed when `All` is selected.

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-4141](https://codice.atlassian.net/browse/DDF-4141)

#### Screenshots
![screen shot 2018-11-07 at 12 54 28 pm](https://user-images.githubusercontent.com/4495447/48157209-4a9e7e80-e28c-11e8-99dc-31cd56c57321.jpg)

#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
